### PR TITLE
[nomenclature] created a nomenclature.md file

### PR DESCRIPTION
**Pull request by robcontreras**
created a nomenclature.md file with some of the most common terms used in cocoapods

### CODE REVIEW

File: NOMENCLATURE.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+### User target
+The target being integrated.
+
+### Workspace
+A workspace is an Xcode document that groups projects and other documents so you can work on them together.
+
+### XCode
+Xcode is an integrated development environment for macOS containing a suite of software development tools developed by Apple for developing software for macOS, iOS, watchOS and tvOS.
+
+### XCodeproj
```
Comment (dnkoutso) [inline NOMENCLATURE.md]: nit use `Xcodeproj` (lower case 'C'`

File: NOMENCLATURE.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+### Umbrella target
+The target generated to aide the integration of the Pod targets with the user targets.
+
+### User target
+The target being integrated.
+
+### Workspace
+A workspace is an Xcode document that groups projects and other documents so you can work on them together.
+
+### XCode
```
Comment (dnkoutso) [inline NOMENCLATURE.md]: also nit `Xcode`

File: NOMENCLATURE.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+## Glossary
+
+### Build
+A package generated after the code in your project has been compiled.
+
+### Bundle
+A bundle is a directory with a standardized hierarchical structure that holds executable code and the resources used by that code.
+
+### Cocoapods
```
Comment (dnkoutso) [inline NOMENCLATURE.md]: nit: `CocoaPods`

File: NOMENCLATURE.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+An organized directory that manages the files to be used in a project along with its dependencies.
+
+### Podfile
+The Podfile is a specification that describes the dependencies of the targets of one or more Xcode projects.
+
+### Repo
+A virtual space where your code is stored.
+
+### Shared Library
+A Shared Library is a Library that is not copied into the project, instead, it's path is referenced by the project.
```
Comment (benasher44) [inline NOMENCLATURE.md]: I'd just remove "path is". I don't think location is important here

File: NOMENCLATURE.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+### Static Library
+A Static Library is a Library that has been copied and bundled into the build.
```
Comment (benasher44) [inline NOMENCLATURE.md]: Replace "has been copied and bundled" with "statically linked" (and maybe link to something about static and dynamic linking?)

File: NOMENCLATURE.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+### Spec
+A specification describes a version of Pod library. It includes details about where the source should be fetched from, what files to use, the build settings to apply, and other general metadata such as its name, version, and description.
+
+### Subproject
+A Subproject is an XCode Project embedded into another XCode project's tree.
+
+### Subspec
+Represents specification for a module of the library.
```
Comment (benasher44) [inline NOMENCLATURE.md]: Nit: replace "of" with "within"

File: NOMENCLATURE.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+A target specifies a product to build and contains the instructions for building the product from a set of files in a project or workspace.
+
+### Trunk
+The CocoaPods Trunk is an authentication and CocoaPods API service used for the distribution of public pods.
+
+### Umbrella target
+The target generated to aide the integration of the Pod targets with the user targets.
+
+### User target
+The target being integrated.
```
Comment (benasher44) [inline NOMENCLATURE.md]: Maybe in parentheses: "part of the user's project outside of CocoaPods"

File: NOMENCLATURE.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+### Workspace
+A workspace is an Xcode document that groups projects and other documents so you can work on them together.
+
+### Xcode
+Xcode is an integrated development environment for macOS containing a suite of software development tools developed by Apple for developing software for macOS, iOS, watchOS and tvOS.
+
+### Xcodeproj
+The XCodeproj is a file generated by XCode that contains contains project configuration, data, and links to files referenced by the project.
```
Comment (benasher44) [inline NOMENCLATURE.md]: 2 places in here: lowercase the c in XCode*

**Comment by robcontreras** [comment]
want to help out more in whatever way i can, i wanted to start with this to actually do something

File: CHANGELOG.md [hunk 9]
```diff
@@ -5203,6 +5203,8 @@ podspec, then update these method calls.
 
 * Updated Xcodeproj to have a friendlier API.
 
+* Created `NOMENCLATURE.md` to keep a glossary of the most common terms used in cocoapods.
```
Comment (benasher44) [inline CHANGELOG.md]: Can this be moved near the top of the CHANGELOG.md? There's a section near the top called master. You can add it to the enhancements section there, and then you can copy the formatting used in previous enhancements sections.

File: NOMENCLATURE.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
+# Cocoapods Nomenclature
```
Comment (benasher44) [inline NOMENCLATURE.md]: nit: Cocoapods -> CocoaPods

File: NOMENCLATURE.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+A list of terms used in the [cocoapods](https://github.com/CocoaPods/CocoaPods) context.
```
Comment (benasher44) [inline NOMENCLATURE.md]: nit: cocoapods -> CocoaPods

File: NOMENCLATURE.md [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+A bundle is a directory with a standardized hierarchical structure that holds executable code and the resources used by that code.
+
+### CocoaPods
+CocoaPods is a dependency manager for Swift and Objective-C Cocoa projects.
+
+### Dependency
+A file that is required by the project to execute a task.
+
+### Dynamically Linked
+When you Dynamically Link a file, a pointer to the file being linked in (the file name of the file, for example) is included in the executable and the contents of said file are not included at link time. It's only when you later run the executable that these dynamically linked files are bought in and they're only bought into the in-memory copy of the executable, not the one on disk.
```
Comment (benasher44) [inline NOMENCLATURE.md]: bought -> brought (2x)

File: NOMENCLATURE.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+### User target
+The target being integrated (part of the user's project outside of CocoaPods).
+
+### Workspace
+A workspace is an Xcode document that groups projects and other documents so you can work on them together.
+
+### Xcode
+Xcode is an integrated development environment for macOS containing a suite of software development tools developed by Apple for developing software for macOS, iOS, watchOS and tvOS.
+
+### Xcodeproj
```
Comment (benasher44) [inline NOMENCLATURE.md]: nit: Xcodeproj -> Xcodeproj File (to disambiguate between the file and the open source project)

**Comment by benasher44** [comment]
@dnkoutso this looks good on my end. Any final thoughts?

**Comment by dnkoutso** [comment]
This is good but I think we should focus more on describing CocoaPods specific terms rather than general iOS/macOS terms that their definition can often found at apple website.

I am a bit hesitant on merging this more on the sense that we will have to support/maintain it. I'd much rather see a document that describes what CocoaPods does in terms of wiring (pod targets, aggregate targets, generated xcconfig, analyzer and resolver, etc.)

I won't block this by any means though!

**Comment by orta** [comment]
Think in terms of nonclamenture - this stuff isn't going to change really. Unless there are some drastic changes coming from Apple. Seems 👍 to me. 

Would also be interested in some of those terms, perhaps they can be added also (can be separate pr)?

**Comment by benasher44** [comment]
@robcontreras thanks for your contribution!

**Pull request merged**